proto/cosi.proto

```diff
@@ -123,15 +153,19 @@ message S3AccessInfo {
     string access_secret_key = 2;
 }
 
-// S3SignatureVersion is the version of the signing algorithm for all s3 requests
-enum S3SignatureVersion {
-    UNKNOWN = 0;
+// S3SignatureVersion is the version of the signing algorithm for all S3 requests
+message S3SignatureVersion {
+    enum Version {
```

Comment:
`.proto` files follow C/C++ rules for enum naming. With the previous version of this enum, UKNOWN only applied to S3SignatureVersion. Having S3SignatureVersion as a message with an encapsulated enum allows for more precise naming. I took this strategy from [CSI's proto definitions](https://github.com/container-storage-interface/spec/blob/master/spec.md)